pkg/runner/dochandlers/v23/v23.go

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package v23
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/opensbom-generator/parsers/meta"
+	"github.com/spdx/spdx-sbom-generator/pkg/runner/options"
+	"github.com/spdx/tools-golang/spdx"
+)
+
+type Handler struct{}
+
+func (h *Handler) CreateDocument(opts *options.Options) (interface{}, error) {
+	doc := &spdx.Document{
+		SPDXVersion: spdx.Version,
+		DataLicense: spdx.DataLicense,
+		// ... this needs to be completed with all the required fuields
+	}
+
+	// TODO: Create top-level package
+	// TODO: Relate top-level package to document
+
+	return doc, nil
+}
+
+func (h *Handler) AddDocumentPackages(opts *options.Options, rawDoc interface{}, metaPackages []meta.Package) error {
+	// The document gets fed as an empty interface, cast it:
+	/*
+		doc, ok := rawDoc.(*spdx.Document)
+		if !ok {
+			return errors.New("unable to cast the document object")
+		}
+	*/
+
+	// TODO: Create the new SPDX packages
+	// TODO: Assign the meta package data
+	// TODO: Add to document
+	// TODO: Relate the package to the top level directory
```

Comment:
@ba11b0y:
This function captures the main conversion of metapackage to spdx 2.3 package. It should implement each step, including your [`convertToPackage()`](https://github.com/opensbom-generator/spdx-sbom-generator/pull/293/files#diff-b1b1ceed0867564530b26682e0ec1ed2d5727b7778ad1c201d43b8ba0b75196aR234-R254) function in your original PR.